### PR TITLE
ci: fix docs changes expected status hang forever

### PR DESCRIPTION
**Pull request by shoothzj**
### Motivation

docs PR hang forever wait for status report, like https://github.com/apache/bookkeeper/pull/4389. The root reason is
> If, however, a job within a workflow is skipped due to a conditional, it will report its status as "Success". For more information, see "[Using conditions to control job execution](https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution)."

See https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks for more details.

Test in my fork repo and it works well.

- Docs change only https://github.com/shoothzj/bookkeeper/pull/5
- Code change https://github.com/shoothzj/bookkeeper/pull/6

Thanks to @tisonkun @waynexia @lhotari

### CODE REVIEW

**Comment by lhotari** [comment]
> We can add same name check to make it pass, I have tested in my fork repo. [shoothzj#3](https://github.com/shoothzj/bookkeeper/pull/3)

I wouldn't recommend such a hack. Why not use the same pattern as there is used in https://github.com/apache/pulsar/blob/master/.github/workflows/pulsar-ci.yaml ?

File: .github/workflows/docs-unittest-status.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      - branch-*
+  push:
+    branches:
+      - master
+      - branch-*
+  workflow_dispatch:
+
+jobs:
+  compatibility-check-java8:
+    if: ${{ needs.build-and-license-check.outputs.docs_only == 'true' }}
```
Comment (lhotari) [inline .github/workflows/docs-unittest-status.yml:36]: this isn't valid currently.

**Review by lhotari — CHANGES_REQUESTED** [review]

**Comment by shoothzj** [comment]
@lhotari ok, let me a take a look later

**Comment by shoothzj** [comment]
@lhotari PTAL

**Review by eolivelli — APPROVED** [review]
+1

**Review by nicoloboschi — APPROVED** [review]
LGTM

File: .github/workflows/bk-ci.yml [hunk 2]
```diff
@@ -533,3 +533,43 @@ jobs:
           name: dependency report
           path: target/dependency-check-report.html
           retention-days: 7
+  bookkeeper-ci-checks-completed:
```
[the remark this answers is not in the document]
Comment (shoothzj) [reply .github/workflows/bk-ci.yml:537]: @lhotari Fixed

**Pull request merged**